### ISSUE TITLE
Right clic to globally disable RAS

### ISSUE THREAD

**Issue by M5oul** [dillbyrne/random-agent-spoofer#245]
In the [0.9.3.1.1 version](https://addons.mozilla.org/en-US/firefox/addon/random-agent-spoofer/) we could globally disable RAS right clicing on the old logo. It was a good feature. Why this feature isn't here with the 0.9.5.3 version?

**Comment by Mosrite** [comment]
+1

**Comment by dillbyrne** [comment]
Hello @M5oul 

The feature was removed for several reasons.

1. RAS needed to have 2 copies of every preference used to maintain state. This meant that it had to change some of the built in pref values on toggle to ensure that they were in the enabled or enabled state. Some of the UI prefs changed more than one about config preference and in the end it confused many users. They complained that RAS should not be changing those prefs after they had changed them. 

2. It was lots of extra code and memory . The toggle based system was at the core of the addon and each preference needed to have its own methods to handle things differently and store the values in the event that the state was changed. Having all that extra code added lots of overhead when adding the simplest of features as methods had to be written for each individual preference to store the value of its current state and then depending on the values the user had chosen, either  set one or reset the preference.

Removing the toggle based system allowed for a rewrite of most of the preference handling code to the point that new options can be added very quickly. Most of the code has been reduced to a few methods that handle the whole lot of it.

3. The widget module has been deprecated and has been replaced in RAS by the action button so this no longer allows a user to left click on the icon and the right click has to be used to show the panel. For a while I did use a mouse menu entry in place to allow the toggle functionality but it was not the same . 


4. I realize that the feature was useful but I think completely disabling RAS on a particular site is worse than unchecking a few boxes or manually changing the profile as you still get some protections that way. Completely disabling it would be the like disabling tor browser when a captcha is presented, obviously they are not on the same scale but I think you get my point.


That being said, I do plan to add something similar where users can choose from presets of preferences  #139 . I want to wait until I get the cookie functionality in place before I begin this. 

I'm not sure yet what way it will be done at this point in time but I will be asking for user input in that issue when it comes to implementing it. Regardless of what ends up being implemented the `ActionButton`  does not allow for multiple clicks so the new feature will either use the context menu or it will use a slider or both. 

I'm closing this issue as the other issue I have open will deal with this.
